### DISCUSSION TITLE
Capturing HTTP headers in java agent

### DISCUSSION THREAD

**Discussion by stanosamek** [open-telemetry/opentelemetry-java-instrumentation#4733] [Q&A] [answered]
Hi, 

I would like to capture some HTTP server request headers, e.g. x-bussiness-user. But, when I configured java agent with parameter:                 -Dotel.instrumentation.common.experimental.capture-http-headers.server.request=http.request.header.x_business_user=[""], In result, I can't see that agent is capturing header. I'm using agent in version 1.7.2.

Where I have problem please? 

Thank you

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @stanosamek,
Try using `-Dotel.instrumentation.common.experimental.capture-http-headers.server.request=x-business-user` (or `-Dotel.instrumentation.http.capture-headers.server.request=x-business-user` if you decide to upgrade to agent 1.9.0) instead. This configuration option accepts a comma-separated list of header names -- not the span attribute names.

**Reply by stanosamek** [reply]
Hi @mateuszrzeszutek,

I'm trying it with suggestion, but without any success. Here is my configuration (I've tried it with '-' or '_') :

              -Dotel.resource.attributes=service.name=foobar,deployment.environment=dit3,namespace_name=example,system_name=omnisvcs
                -Dotel.exporter.otlp.endpoint='http://otel-collector.opentelemetry.svc.cluster.local:4317'
                -Dotel.instrumentation.http.capture-headers.client.request=x-business-user,x-process-id,x-dynatrace
                -Dotel.instrumentation.http.capture-headers.client.response=x-business-user,x-process-id,x-dynatrace
                -Dotel.instrumentation.common.experimental.capture-http-headers.server.request=x-business-user,x-process-id,x-dynatrace
                -javaagent:'/var/run/otel/opentelemetry-javaagent-all.jar'

Do you have any suggestion how can I debug the issue or can be problem?

**Reply by stanosamek** [reply]
Hi @mateuszrzeszutek,

I turn on debug level on Collector side and now I can see spans contains attributes from headers. I can't see them on our storage but that will be problem on our side. Thank you for your help.
